packages/client/hmi-client/src/services/mit-askem.ts

```diff
@@ -0,0 +1,30 @@
+import API, { Poller } from '@/api/api';
+import { logger } from '@/utils/logger';
+
+export async function codeToAcset(code: string) {
+	const response = await API.post(`code/to_acset`, code);
+	return response.data ?? null;
+}
+
+export async function findVarsFromText(text: string) {
+	const id = await API.post(`/code/annotation/find_text_vars`, text);
+	const poller = new Poller<object>()
+		.setInterval(2000)
+		.setThreshold(90)
+		.setPollAction(async () => {
+			const response = await API.get(`/code/response?id=${id.data}`);
+			if (response) {
+				return {
+					data: response.data.data
+				};
```

Comment:
Since response is not null regardless of success or error, it is probably prudent to be more pedantic here, both to guard against errors and to convey intention.

```
const { status, data } = response;
if (status >= 400) return { data: null, progress: null, error: "failed" };
return { data: data.data, progress: null, error: null };
```

Reply:
The response is null actually, otherwise the poller wouldn't work at all!

Reply:
Look like the hmi-server resource is not catching the errors. Will patch this in a different PR.